### DISCUSSION TITLE
filesService import URL from disk

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#15787] [Archive]
As far as I know it is not possible to import a file from a local disk?

I am trying to import a file which is next to directus automaticly.
My idea was to use the filesService.importOne

The URL: `file:///directus/file.txt` is not accecpted same as `/directus/file.txt`

Am i missing something?

**Comment by NilsBaumgartner1994** [comment]
Okay probably, it is because the url is forbidden by default:
```
...env.IMPORT_IP_DENY_LIST.includes(ip)...
```

**Reply by NilsBaumgartner1994** [reply]
https://github.com/directus/directus/blob/main/api/src/services/files.ts line 237

**Comment by azrikahar** [comment]
`importOne` is meant for remote URLs, particularly with the usage of axios, so AFAIK it is much different from the use case you intended:

https://github.com/directus/directus/blob/cbe1e7fbc69c8e9bd02254e9ed6efc35ba6468ef/api/src/services/files.ts#L247-L249

I believe the technically correct solution would be to read the local file and pass it to `uploadOne` as a readable stream:

https://github.com/directus/directus/blob/cbe1e7fbc69c8e9bd02254e9ed6efc35ba6468ef/api/src/services/files.ts#L33-L38

example usage of a similar use case for file exports that are saved to the File Library:

https://github.com/directus/directus/blob/cbe1e7fbc69c8e9bd02254e9ed6efc35ba6468ef/api/src/services/import-export.ts#L264

**Reply by NilsBaumgartner1994** [reply]
You were very quick ;-) Figured this also out. Will mark this, since it is the same way i did it.
